sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -66,10 +71,33 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
             }
           case _ => None
         }
+      case (resExp, r @ DataSourceV2ScanRelation(_, scan: SupportsDynamicFiltering, _)) =>
+        val filterAttrs = AttributeSet(resolveRefs(r, scan.filterAttributes))
```

Comment:
What if some attributes in `scan.filterAttributes` cannot be resolved? should we skip and continue?
Also you may want to use `DataSourceV2ScanRelation.relation` since the output could be pruned by column pruning. It seems we currently run `PartitionPruning` after `V2ScanRelationPushDown`.

Reply:
+1 on skipping rather than failing if we cannot resolve the filter attrs.

W.r.t. which columns to use, I did use the scan output to resolve on purpose as we cannot derive a filter on an attribute that hasn't been projected.

Reply:
@sunchao, it looks like we always fail if partition columns cannot be resolved in the branch above this one. While it does seem safer not to fail the query if a filter attribute cannot be resolved, shall we be consistent with v1 tables? What do you think?

Reply:
Yeah, SGTM.